.dotnet/src/Custom/Chat/ChatCompletionOptions.cs

```diff
@@ -159,4 +159,10 @@ public ChatCompletionOptions()
     // CUSTOM: Added the Obsolete attribute.
     [Obsolete($"This property is obsolete. Please use {nameof(Tools)} instead.")]
     public IList<ChatFunction> Functions { get; }
+
+    [CodeGenMember("Metadata")]
+    public IDictionary<string, string> Metadata { get; } = new ChangeTrackingDictionary<string, string>();
```

Comment:
I can see that the "store" property needs to be renamed, but why does "metadata" need to be customized? 🤔 Could you add `// CUSTOM` comments for both properties?

Reply:
Comments added. Metadata is customized to remove the public setter it otherwise has emitted for it.